examples/js/loaders/GLTFLoader.js

```diff
@@ -733,6 +744,110 @@ THREE.GLTFLoader = ( function () {
 
 	};
 
+	/**
+	 * Materials variants extension
+	 *
+	 * Specification: https://github.com/KhronosGroup/glTF/tree/master/extensions/2.0/Khronos/KHR_materials_variants
+	 */
+	function GLTFMaterialsVariantsExtension( parser ) {
+
+		this.parser = parser;
+		this.name = EXTENSIONS.KHR_MATERIALS_VARIANTS;
+		this.addExtensionsToUserData = true;
+
+	}
+
+	GLTFMaterialsVariantsExtension.prototype.afterRoot = function ( gltf ) {
+
+		var parser = this.parser;
+		var json = parser.json;
+		var name = this.name;
+
+		if ( ! json.extensions || ! json.extensions[ name ] ) return null;
+
+		var extensionDef = json.extensions[ name ];
+		var variantsDef = extensionDef.variants || [];
+		var variants = [];
+
+		for ( var i = 0, il = variantsDef.length; i < il; i ++ ) {
+
+			variants.push( variantsDef[ i ].name );
+
+		}
+
+		for ( var i = 0, il = gltf.scenes.length; i < il; i ++ ) {
+
+			gltf.scenes[ i ].userData.variants = variants;
+
+		}
```

Comment:
Note: I save variants in `scene.userData.variants`, too, because they can be used for future glTF exporter support. Otherwise, variants parameter may need to be separately passed to the exporter.